llvm/test/CodeGen/AArch64/zext.ll

```diff
@@ -1089,67 +1089,35 @@ entry:
 }
 
 define <16 x i16> @zext_v16i10_v16i16(<16 x i10> %a) {
-; CHECK-SD-LABEL: zext_v16i10_v16i16:
-; CHECK-SD:       // %bb.0: // %entry
-; CHECK-SD-NEXT:    ldr w8, [sp]
-; CHECK-SD-NEXT:    fmov s0, w0
-; CHECK-SD-NEXT:    ldr w9, [sp, #8]
-; CHECK-SD-NEXT:    fmov s1, w8
-; CHECK-SD-NEXT:    ldr w8, [sp, #16]
-; CHECK-SD-NEXT:    mov v0.h[1], w1
-; CHECK-SD-NEXT:    mov v1.h[1], w9
-; CHECK-SD-NEXT:    mov v0.h[2], w2
-; CHECK-SD-NEXT:    mov v1.h[2], w8
-; CHECK-SD-NEXT:    ldr w8, [sp, #24]
-; CHECK-SD-NEXT:    mov v0.h[3], w3
-; CHECK-SD-NEXT:    mov v1.h[3], w8
-; CHECK-SD-NEXT:    ldr w8, [sp, #32]
-; CHECK-SD-NEXT:    mov v0.h[4], w4
-; CHECK-SD-NEXT:    mov v1.h[4], w8
-; CHECK-SD-NEXT:    ldr w8, [sp, #40]
-; CHECK-SD-NEXT:    mov v0.h[5], w5
-; CHECK-SD-NEXT:    mov v1.h[5], w8
-; CHECK-SD-NEXT:    ldr w8, [sp, #48]
-; CHECK-SD-NEXT:    mov v0.h[6], w6
-; CHECK-SD-NEXT:    mov v1.h[6], w8
-; CHECK-SD-NEXT:    ldr w8, [sp, #56]
-; CHECK-SD-NEXT:    mov v0.h[7], w7
-; CHECK-SD-NEXT:    mov v1.h[7], w8
-; CHECK-SD-NEXT:    bic v0.8h, #252, lsl #8
-; CHECK-SD-NEXT:    bic v1.8h, #252, lsl #8
-; CHECK-SD-NEXT:    ret
-;
-; CHECK-GI-LABEL: zext_v16i10_v16i16:
-; CHECK-GI:       // %bb.0: // %entry
-; CHECK-GI-NEXT:    fmov s4, w0
-; CHECK-GI-NEXT:    fmov s5, w4
-; CHECK-GI-NEXT:    ldr s0, [sp]
-; CHECK-GI-NEXT:    ldr s1, [sp, #8]
-; CHECK-GI-NEXT:    ldr s2, [sp, #32]
-; CHECK-GI-NEXT:    ldr s3, [sp, #40]
-; CHECK-GI-NEXT:    adrp x8, .LCPI52_0
-; CHECK-GI-NEXT:    mov v4.s[1], w1
-; CHECK-GI-NEXT:    mov v5.s[1], w5
-; CHECK-GI-NEXT:    mov v0.s[1], v1.s[0]
-; CHECK-GI-NEXT:    mov v2.s[1], v3.s[0]
-; CHECK-GI-NEXT:    ldr s1, [sp, #16]
-; CHECK-GI-NEXT:    ldr s3, [sp, #48]
-; CHECK-GI-NEXT:    mov v4.s[2], w2
-; CHECK-GI-NEXT:    mov v5.s[2], w6
-; CHECK-GI-NEXT:    mov v0.s[2], v1.s[0]
-; CHECK-GI-NEXT:    mov v2.s[2], v3.s[0]
-; CHECK-GI-NEXT:    ldr s1, [sp, #24]
-; CHECK-GI-NEXT:    ldr s3, [sp, #56]
-; CHECK-GI-NEXT:    mov v4.s[3], w3
-; CHECK-GI-NEXT:    mov v5.s[3], w7
-; CHECK-GI-NEXT:    mov v0.s[3], v1.s[0]
-; CHECK-GI-NEXT:    mov v2.s[3], v3.s[0]
-; CHECK-GI-NEXT:    ldr q3, [x8, :lo12:.LCPI52_0]
-; CHECK-GI-NEXT:    uzp1 v1.8h, v4.8h, v5.8h
-; CHECK-GI-NEXT:    uzp1 v2.8h, v0.8h, v2.8h
-; CHECK-GI-NEXT:    and v0.16b, v1.16b, v3.16b
-; CHECK-GI-NEXT:    and v1.16b, v2.16b, v3.16b
-; CHECK-GI-NEXT:    ret
+; CHECK-LABEL: zext_v16i10_v16i16:
```

Comment:
Why are we now falling back for this test?

Reply:
Due to the nature in which the parameters are loaded for this function, it gets created as a `v16i32` buildvector.
The parameter then gets truncated to `v16i10`, before being extended to `v16i16`.